File: back/src/main/java/com/julien/plop/template/domain/Template.java
package com.julien.plop.template.domain;

import com.julien.plop.board.model.Board;
import com.julien.plop.scenario.Scenario;

import java.util.UUID;

public record Template(Id id, String code, String label, String version, Scenario scenario, Board board) {

    public record Id(String value) {

    }

    public static Id generateId() {
        return new Id(UUID.randomUUID().toString());
    }

}

File: mobile/lib/contexts/template/domain/template.dart

class Template {

  final String _id;
  final String _code;
  final String _label;

  Template({
    required String id,
    required String code,
    required String label
  }): _id = id, _code = code, _label = label;

  String get id => _id;

  String get code => _code;

  String get label => _label;

}